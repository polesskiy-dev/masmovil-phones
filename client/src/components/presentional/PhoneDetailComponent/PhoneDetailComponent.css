.phone-detail {
    display: flex;
    flex-direction: column;
}

.phone-detail .phone-detail__image {
    max-height: 256px;
    object-fit: contain;

    padding: 0 16px;
}

.phone-detail .phone-detail__description {
    margin-top: 16px;
}