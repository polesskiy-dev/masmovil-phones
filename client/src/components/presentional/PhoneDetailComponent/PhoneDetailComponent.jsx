import React from 'react';
import PropTypes from 'prop-types';
import { FormattedMessage } from 'react-intl';

import './PhoneDetailComponent.css';

const PhoneDetailComponent = ({
  DeviceName,
  img,
  type,
  size,
  technology,
  announced,
}) => (
  <article className="phone-detail">
    <img className="phone-detail__image" src={img} alt={DeviceName} />
    <dl className="phone-detail__description">
      <dt>
        <FormattedMessage id="Phone.Type" />
      </dt>
      <dd>{type}</dd>
      <dt>
        <FormattedMessage id="Phone.Size" />
      </dt>
      <dd>{size}</dd>
      <dt>
        <FormattedMessage id="Phone.Technology" />
      </dt>
      <dd>{technology}</dd>
      <dt>
        <FormattedMessage id="Phone.Announced" />
      </dt>
      <dd>{announced}</dd>
    </dl>
  </article>
);

PhoneDetailComponent.propTypes = {
  DeviceName: PropTypes.string.isRequired,
  img: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  size: PropTypes.string.isRequired,
  technology: PropTypes.string.isRequired,
  announced: PropTypes.string.isRequired,
};

export default PhoneDetailComponent;
