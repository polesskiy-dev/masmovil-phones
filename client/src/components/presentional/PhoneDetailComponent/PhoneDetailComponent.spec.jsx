import React from 'react';
import renderer from 'react-test-renderer';

import withDefaultIntl from '../../../helpers/with-default-intl';
import PhoneDetailComponent from './PhoneDetailComponent';

it('<PhoneDetailComponent/ > renders correctly with all required fields', () => {
  const tree = renderer
    .create(
      withDefaultIntl(
        <PhoneDetailComponent
          type=""
          announced=""
          DeviceName=""
          img=""
          size=""
          technology=""
        />
      )
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
