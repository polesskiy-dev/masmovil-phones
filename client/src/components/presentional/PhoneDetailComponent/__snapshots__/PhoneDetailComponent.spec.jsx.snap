// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PhoneDetailComponent/ > renders correctly with all required fields 1`] = `
<article
  className="phone-detail"
>
  <img
    alt=""
    className="phone-detail__image"
    src=""
  />
  <dl
    className="phone-detail__description"
  >
    <dt>
      <span>
        Type
      </span>
    </dt>
    <dd>
      
    </dd>
    <dt>
      <span>
        Size
      </span>
    </dt>
    <dd>
      
    </dd>
    <dt>
      <span>
        Technology
      </span>
    </dt>
    <dd>
      
    </dd>
    <dt>
      <span>
        Announced
      </span>
    </dt>
    <dd>
      
    </dd>
  </dl>
</article>
`;
